# References

Undestanding Chi-Square test: https://stattrek.com/chi-square-test/independence.aspx
https://www.investopedia.com/terms/c/chi-square-statistic.asp#:~:text=A%20chi%2Dsquare%20(%CF%872,compares%20to%20actual%20observed%20data.&text=The%20chi%2Dsquare%20statistic%20compares,of%20variables%20in%20the%20relationship)
Understanding desgrees of freedom:https://statisticsbyjim.com/hypothesis-testing/degrees-freedom-statistics/
Scipy docs:https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.chisquare.html


# Table

In [5]:
headings = ["A","B","C","D"]
data =[[ 90, 60, 104, 95],[ 30, 50, 51, 20],[30, 40, 45, 35]]
format_row = "{:>10}" * (len(headings) + 1)
print(format_row.format("", *headings))
for team, row in zip(headings, data):
    print(format_row.format(team, *row)) 

                   A         B         C         D
         A        90        60       104        95
         B        30        50        51        20
         C        30        40        45        35


# Code

In [12]:
import scipy.stats as stats
  
# data stored in a 2-D array from the table provided
data = [[ 90, 60, 104, 95],[ 30, 50, 51, 20],[30, 40, 45, 35]]
#set a significant value
sigValue = 24.6
stat, p, dof, expected = stats.chi2_contingency(data) 

print("p value is " + str(p)) 

print("Expected")
print(expected)

print("Dof = ", dof)
print("Stat = ", stat)


p value is 0.0004098425861096696
Expected
[[ 80.53846154  80.53846154 107.38461538  80.53846154]
 [ 34.84615385  34.84615385  46.46153846  34.84615385]
 [ 34.61538462  34.61538462  46.15384615  34.61538462]]
Dof =  6
Stat =  24.5712028585826


# Analysis

The result received in this test is 24.57. This is lower than the significant value of 24.6. The difference is 0.03.
A null hypothesis can now not be accepted because we can see that there is a relationship between the 2 categories.
The value we get for p also reaffirms this conclusion.